### PR TITLE
Epic/cv2 5050 text vectorization via presto

### PR DESCRIPTION
**Pull request by DGaffney**
## Description

Epic branch roll-up of all the changes required to shift to presto-based text vectorization. 

References: CV2-5050

## How has this been tested?

Tested extensively locally and in integration tests. 

## Things to pay attention to during code review

Help me make sure we're not accidentally bypassing some important rule-based logic for matches!

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

**Review by caiosba — COMMENTED** [review]
Is it all ready to be merged with `develop`, Devin? :thinking:

**Comment by DGaffney** [comment]
> Is it all ready to be merged with `develop`, Devin? 🤔

Sorry, no, I was opening this to test something - will set to draft!

**Review by caiosba — CHANGES_REQUESTED** [review]
@DGaffney , thanks for all the work here! Can you please make sure that we have tests that cover these lines (or confirm if these lines should be removed, or if the fact that these lines are not covered means that there is something wrong in the code paths): 

- `app/models/concerns/alegre_v2.rb`: 41, 325, 326 and 327

File: app/models/concerns/alegre_v2.rb [hunk 1]
```diff
@@ -148,28 +163,32 @@ def get_type(project_media)
...
     def content_hash(project_media, field)
       if Bot::Alegre::ALL_TEXT_SIMILARITY_FIELDS.include?(field)
-        Digest::MD5.hexdigest(project_media.send(field))
+        content_hash_for_value(project_media.send(field))
+      elsif project_media.is_link?
+        return content_hash_for_value(project_media.media.url)
+      elsif project_media.is_a?(TemporaryProjectMedia)
+        return Rails.cache.read("url_sha:#{project_media.url}")
+      elsif project_media.is_uploaded_media?
+        return project_media.media.file.filename.split(".").first
```
Comment (caiosba) [inline app/models/concerns/alegre_v2.rb:178]: @DGaffney how strict the content hash should be? Do you expect no clashes? Just noting here that two upload files can have the same name but completely different contents.

**Review by caiosba — COMMENTED** [review]
Thanks @DGaffney for completing the 100% code coverage - no many changes since my last review, so this time I just left a question for something I didn't think about before.

File: app/models/concerns/alegre_v2.rb (hunk 1 above)
Comment (DGaffney) [reply app/models/concerns/alegre_v2.rb:178]: I think your example would be ok - if a file has a different context, but an existing content hash, I think that's... actually the ideal case? we'd side-step fingerprinting based on the fact we'd seen that media before and just add context to that case? Maybe something to pin as an explicit test for when we roll this out on QA?

Comment (caiosba) [reply app/models/concerns/alegre_v2.rb:178]: Sorry, content, not context :D

Comment (caiosba) [reply app/models/concerns/alegre_v2.rb:178]: As per conversation with @computermacgyver and @DGaffney , we confirmed filenames are hashes of the content: https://github.com/meedan/check-api/blob/develop/app/models/media.rb#L35-L41. @DGaffney will just do a last confirmation: Upload two files with the same name but different content and be sure that they have different content hashes.

Comment (DGaffney) [reply app/models/concerns/alegre_v2.rb:178]: @caiosba confirmed and noted as such on the PR!

Comment (caiosba) [reply app/models/concerns/alegre_v2.rb:178]: Thanks!

File: app/workers/reindex_alegre_workspace.rb [hunk 2]
```diff
@@ -64,11 +67,14 @@ def get_request_docs_for_project_media(pm)
     end
   end
 
-  def check_for_write(running_bucket, event_id, team_id, write_remains=false, in_processes=3)
+  def check_for_write(running_bucket, event_id, team_id, write_remains=false, _in_processes=3)
```
Comment (melsawy) [inline app/workers/reindex_alegre_workspace.rb:70]: Q: Why do we need to keep the last argument `_in_processes`? I think we can remove it.

File: app/models/concerns/project_media_getters.rb [hunk 3]
```diff
@@ -31,6 +31,10 @@ def is_image?
     self.is_uploaded_image?
```
Comment (melsawy) [inline app/models/concerns/project_media_getters.rb:31]: It's better to remove `is_uploaded_image?` method and move the logic inside `is_image?` method for consistent 
I know this one not related to your PR but I suggest to include this refactor.

**Review by melsawy — APPROVED** [review]
Great work @DGaffney 
I left a small refactor suggestions, please check my comments.

**Review by jayjay-w — APPROVED** [review]

**Pull request merged**